### ISSUE TITLE
Unlike d3.rgb, d3.hsl doesn’t special case a brighter black.

### ISSUE THREAD

**Issue by p3k** [mbostock/d3#1815]
[Original title: d3.hsl('#000000') evaluates to NaN for hue and saturation]

If I create an HSL object from the color string '#000000' (black) it comes with NaN values for its h and s properties:

```javascript
> d3.hsl('#000000')
d3_Hsl {h: NaN, s: NaN, l: 0, brighter: function, darker: function…}
> d3.version
"3.4.4"
```

I would expect something like `{h: 0, s: 0, l: 0}` instead…

In consequence, I am not getting a lighter color (gray) when issuing e.g. `d3.hsl('#000000').brighter()` – the result remains the same with the NaN values.

**Comment by mbostock** [comment]
FWIW, in d3-color neither RGB or HSL special-cases black. So, brighter has no effect on black in these color spaces. (It works fine in Lab and HCL, because there brighter and darker are additive rather than multiplicative.)

**Comment by mbostock** [comment]
Closing as fixed in d3-color for 4.0.
